tiledb/sm/query/query.cc

```diff
@@ -2203,7 +2204,8 @@ Status Query::set_subarray_unsafe(const NDRange& subarray) {
 
 Status Query::check_buffers_correctness() {
   if (type_ != QueryType::READ && type_ != QueryType::WRITE &&
-      type_ != QueryType::MODIFY_EXCLUSIVE && type_ != QueryType::DELETE) {
+      type_ != QueryType::MODIFY_EXCLUSIVE && type_ != QueryType::DELETE &&
+      type_ != QueryType::UPDATE) {
     return LOG_STATUS(Status_SerializationError(
         "Cannot check buffers; Unsupported query type."));
   }
```

Comment:
Optional: This might be a good opportunity to add an `ensure_query_type_is_valid` method to query_type.h.